### DISCUSSION TITLE
Celery, Flask, and python logging

### DISCUSSION THREAD

**Discussion by jessielw** [pallets/flask#4975] [Q&A]
This probably is more of a celery issue over flask however, I found lots of things online for django but none really for flask when it came to this issue. 

The issue is when I'm using the logging module to log messages etc, it also makes a duplicate celery log at the same time. 

I found this example on SO [here](https://stackoverflow.com/questions/45340837/prevent-duplicate-celery-logging)

```python
from celery import Celery

import logging

app = Celery('tasks', broker='redis://localhost:6379/0')

# logging setup
bhd_logger = logging.getLogger("bhdstudio_site")
bhd_logger.setLevel(logging.INFO)
_formatter = logging.Formatter("%(name)s:%(asctime)s:%(levelname)s = %(message)s")
_stream_handler = logging.StreamHandler()
_stream_handler.setFormatter(_formatter)
bhd_logger.addHandler(_stream_handler)
# bhd_logger.propagate = False

def foo():
    bhd_logger.info("log info from foo")
```

**This will output 'log info from foo' in both the python logging module AND celery without bhd_logger.propagate = False**
    
The solution was to set `bhd_logger.propagate = False` in this 5 year old thread.  I was wondering if this was the recommended way? As this does seem to work as long as the python logger/celery logger does not sit on the same log level as celery. It seems to dead lock if I was to swap celery to the same logging level with this setup.

**Comment by ThiefMaster** [comment]
IMHO anything but very small applications shouldn't use `app.logger` at all but just use python's logging module directly

**Reply by jessielw** [reply]
Sorry since the post probably didn't make this clear. However, I am using the python logger to handle logging and not the app logger.

**Reply by jessielw** [reply]
I edited the OP with the logging setup I'm using
